common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -0,0 +1,113 @@
+"""Simple utility functions that operate on course metadata.
+
+Ths is a place to put simple functions that operate on course metadata. It
+allows us to share code between the CourseDescriptor and CourseOverview
+classes, which both need these type of functions.
+"""
+
+from datetime import datetime
+from base64 import b32encode
+
+from django.utils.timezone import UTC
+
+from .fields import Date
+
+DEFAULT_START_DATE = datetime(2030, 1, 1, tzinfo=UTC())
+
+def clean_id(location, padding_char):
+    """Returns a unique deterministic base32-encoded ID for a course.
+
+    Args:
+        location (UsageKey): The UsageKey of said course
+        padding_char (str): Character used for padding at end of the encoded string.
+                            The standard value for this is '='.
+    """
+    return "course_{}".format(
+        b32encode(unicode(location.course_key)).replace('=', padding_char)
+    )
+
+def display_name_with_default(display_name, location):
+    """Calculate the display name for a course.
+
+    Default to the given display_name if it isn't None, else fall back to creating
+    a name based on the URL (location.name)
+
+    Args:
+        display_name (str): Display name of said course
+        location (UsageKey): Usage key of said course
+    """
+    return (display_name if display_name is not None else location.name.replace('_', ' ')) \
```

Comment:
Rather than using `\`, wrap the code in parenthesis `()`.  For example:

```
(
    display_name if display_name is not None else location.name.replace('_', ' ')
).replace('<', '&lt;').replace('>', '&gt;')
```